Dockerfile

```diff
@@ -1,52 +1,39 @@
-## semgrep build
-
-FROM python:3.7.7-alpine3.11 as build-semgrep
-RUN apk add --no-cache git
-COPY .git /home/pythonbuild/.git
-COPY semgrep /home/pythonbuild/semgrep/
-WORKDIR /home/pythonbuild/semgrep
-# downgrade to 20.1 since 20.1.1 reverts building in place
-# https://github.com/pypa/pip/issues/7555
-RUN pip install pip==20.1
-RUN HOMEBREW_SYSTEM='NOCORE' pip install --user .
-RUN /root/.local/bin/semgrep --version
-
-## semgrep-core build
+# semgrep-core build
 
 FROM ocaml/opam2:alpine@sha256:4c2ce9a181b4b12442a68fc221d0b753959ec80e24eae3bf788eeca4dcb9a293 as build-semgrep-core
+
 USER root
 RUN apk add --no-cache perl m4
-
 USER opam
 
 WORKDIR /home/opam/opam-repository
 RUN git pull && opam update && opam switch create 4.10.0+musl+static+flambda
 
-COPY --chown=opam . /home/opam/semgrep/
```

Comment:
`COPY`ing _everything_ in significantly slows down the build process because the Docker context is huge. I.e. if _anything_ changes in the working directory we have to rebuild `semgrep-core` from scratch, which takes a long time. Now we only rebuild if anything in `.git`, `pfff`, `semgrep-core`, or `.gitmodules` changes, which helpfully avoids the rebuild if `semgrep/` changes.